File: src/typescript/examples/streaming.ts
import { GeminiClient, EventType } from '../src/index.js';

async function main(): Promise<void> {
  console.log('GeminiSDK TypeScript - Streaming Example\n');

  const client = new GeminiClient();

  try {
    await client.start();

    const session = await client.createSession({
      model: 'gemini-2.5-flash',
      streaming: true,
    });

    // Register event handler for streaming
    session.on((event) => {
      switch (event.type) {
        case EventType.ASSISTANT_MESSAGE_DELTA:
          process.stdout.write((event.data as any).deltaContent || '');
          break;
        case EventType.ASSISTANT_MESSAGE:
          console.log('\n--- Complete ---');
          break;
        default:
          break;
      }
    });

    await session.send({
      prompt: 'Write a haiku about TypeScript programming',
    });

    await client.close();
  } catch (error) {
    console.error('Error:', error);
    process.exit(1);
  }
}

main();

File: src/cpp/examples/streaming.cpp
/**
 * @file streaming.cpp
 * @brief Streaming example for GeminiSDK C++
 */

#include <geminisdk/geminisdk.hpp>
#include <iostream>

using namespace geminisdk;

int main() {
    std::cout << "GeminiSDK C++ - Streaming Example\n\n";
    
    try {
        Client client;
        client.start();
        
        SessionConfig config;
        config.model = "gemini-2.5-flash";
        config.streaming = true;
        
        auto session = client.create_session(config);
        
        // Register event handler for streaming
        session->on([](const SessionEvent& event) {
            switch (event.event_type) {
                case EventType::AssistantMessageDelta:
                    if (event.data.contains("deltaContent")) {
                        std::cout << event.data["deltaContent"].get<std::string>() << std::flush;
                    }
                    break;
                case EventType::AssistantMessage:
                    std::cout << "\n--- Complete ---\n";
                    break;
                default:
                    break;
            }
        });
        
        MessageOptions options;
        options.prompt = "Write a haiku about C++ programming";
        
        session->send(options);
        
        client.close();
        
    } catch (const std::exception& e) {
        std::cerr << "Error: " << e.what() << "\n";
        return 1;
    }
    
    return 0;
}
